plugins/call_button/call_button.tsx

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {CSSProperties, useState, useEffect, useRef} from 'react';
+import {injectIntl, IntlShape} from 'react-intl';
```

Comment:
why not use `useIntl`?

Reply:
No idea, the code was (for the most part) already there, I just moved it around and changed the necessary parts. How do these differ?

Reply:
well, it is cleaner as it just adds a hook and does not wrap the component in a HOC

Reply:
yeah, with `useIntl` you don't need to have a prop for it, so that in itself is better. You typically use it like this:

```tsx
const {formatMessage} = useIntl();
```

`formatMessage` is the same function that you are already destructuring from the props, so there is no need to change that part.